### PR TITLE
Force git's hash to be 10 digits max, as per requested by the python script (git 2.34.1 returns a 12 digits by default for --short).

### PR DESCRIPTION
**Pull request by f1rmb**
BTW, default Makefile enables too much stuff, and .text section overflows the flash region. (I disabled MDC1200 locally).

### CODE REVIEW

**Comment by OneOfEleven** [comment]
> BTW, default Makefile enables too much stuff, and .text section overflows the flash region. (I disabled MDC1200 locally).

oh.

I wonder which compiler/version you're using ?

I know it's very close to the limit again but it's OK here when I compile. I'm working on the MDC1200 stuff at the moment, but yes everyone may as well disable it till I have it fixed.

**Comment by f1rmb** [comment]
I'm running LinuxMint, and gcc is

```
arm-none-eabi-gcc (15:10.3-2021.07-4) 10.3.1 20210621 (release)
```

About the flash limit, if I revert my latest PR, I have the error:
```
/usr/lib/gcc/arm-none-eabi/10.3.1/../../../arm-none-eabi/bin/ld: firmware section `.text' will not fit in region `FLASH'
/usr/lib/gcc/arm-none-eabi/10.3.1/../../../arm-none-eabi/bin/ld: region `FLASH' overflowed by 460 bytes
```

now the **main** branch compiles (after I crop the git hash to 10 digits).

**Comment by f1rmb** [comment]
free space looks to be super tight.

File: Makefile [hunk 1]
```diff
@@ -77,7 +77,7 @@ ENABLE_KEYLOCK                   := 1
 
 TARGET = firmware
 
-GIT_HASH_TMP := $(shell git rev-parse --short HEAD)
```
Comment (DualTachyon) [inline Makefile:80]: On the official OEFW, this is actually 7.

Comment (f1rmb) [reply Makefile:80]: Hence the python script has also to be changed (as it checks for a maxlen of 10).

**Pull request closed without merging**